webapp/channels/src/components/admin_console/ip_filtering/edit_section.tsx

```diff
@@ -0,0 +1,197 @@
+
+import React, {useState} from 'react';
+import {Button, Tooltip} from 'react-bootstrap';
+import {useIntl} from 'react-intl';
+
+import {AlertOutlineIcon, PencilOutlineIcon, TrashCanOutlineIcon} from '@mattermost/compass-icons/components';
+import type {AllowedIPRange} from '@mattermost/types/config';
+
+import IPFilteringEarthSvg from 'components/common/svg_images_components/ip_filtering_earth_svg';
+import OverlayTrigger from 'components/overlay_trigger';
+
+type EditSectionProps = {
+    ipFilters: AllowedIPRange[] | null;
+    currentUsersIP: string | null;
+    currentIPIsInRange: boolean;
+    setShowAddModal: (show: boolean) => void;
+    setEditFilter: (filter: AllowedIPRange) => void;
+    handleConfirmDeleteFilter: (filter: AllowedIPRange) => void;
+};
+
+const EditSection: React.FC<EditSectionProps> = ({
+    ipFilters,
+    currentUsersIP,
+    setShowAddModal,
+    setEditFilter,
+    handleConfirmDeleteFilter,
+    currentIPIsInRange,
+}) => {
+    const {formatMessage} = useIntl();
+    const [hoveredRow, setHoveredRow] = useState<number | null>(null);
+
+    const editTooltip = <Tooltip id='edit-tooltip'>{formatMessage({id: 'admin.ip_filtering.edit', defaultMessage: 'Edit'})}</Tooltip>;
+    const deleteTooltip = <Tooltip id='delete-tooltip'>{formatMessage({id: 'admin.ip_filtering.delete', defaultMessage: 'Delete'})}</Tooltip>;
+
+    const ipNotInRangeErrorPanel = (
+        <div className='NotInRangeErrorPanel'>
+            <div className='Icon'>
+                <AlertOutlineIcon size={20}/>
+            </div>
+            <div className='Content'>
+                <div className='Title'>
+                    {formatMessage({id: 'admin.ip_filtering.your_current_ip_is_not_in_allowed_rules', defaultMessage: 'Your IP address {ip} is not included in your allowed IP address rules.'}, {ip: currentUsersIP})}
+                </div>
+                <div className='Body'>
+                    {formatMessage({id: 'admin.ip_filtering.include_your_ip', defaultMessage: 'Include your IP address in at least one of the rules below to continue.'})}
+                    <div
+                        className='Button'
+                        onClick={() => {
+                            setShowAddModal(true);
+                        }}
+                    >
+                        {formatMessage({id: 'admin.ip_filtering.add_your_ip', defaultMessage: 'Add your IP address'})}
+                    </div>
+                </div>
+            </div>
+        </div>
+    );
+
+    function handleRowMouseEnter(index: number) {
+        setHoveredRow(index);
+    }
+
+    function handleRowMouseLeave() {
+        setHoveredRow(null);
+    }
+
+    return (
+        <div className='EditSection'>
+            <div className='AllowedIPAddressesSection'>
+                <div className='SectionHeaderContent'>
+                    <div className='Frame1281'>
+                        <div className='TitleSubtitle'>
```

Comment:
Copy from Figma I forgot to update

Reply:
We might want to be careful about leaning too much on that since I think there's likely cases where there's more semantic markup that we should be using to make sure stuff is accessible and behaves correctly

Reply:
For sure. It's great for getting the basic layout on the page so you can work on back-of-front-end but needs optimizations afterwards